src/client.rs

```diff
@@ -543,6 +546,13 @@ impl ControlChannelHandle {
                         panic!("{:#}. Break", err);
                     }
 
+                    retries += 1;
```

Comment:
I think we actually want to only abort the program when failures are in line. For example, three retry failures in line and abort the program. But your change seems to limit the total retry failures. It's common for long-running instances to have more retry failures because of network outrage. Say it runs for an hour, then fails, and runs for another hour, etc. In that case we don't want to limit the total retry failures, or we can only run for three hours. That said, you want to limit the retry failures in a similar style with the `retry_backoff`